### ISSUE TITLE
Build pushing images for local cluster in 1.17.2

### ISSUE THREAD

**Issue by braderhart** [GoogleContainerTools/skaffold#5176]
### Expected behavior
The following worked previously, where it would not automatically push the images.

```yaml
profiles:
- activation:
  - command: build
    kubeContext: docker-desktop
  build:
    local:
      useBuildkit: true
  name: docker-desktop
```

### Actual behavior
It is now pushing images, unless I explicitly set `push` to `false`:

```yaml
profiles:
- activation:
  - command: build
    kubeContext: docker-desktop
  build:
    local:
      push: false
      useBuildkit: true
  name: docker-desktop
```

### Information

- Skaffold version: `1.17.2`
- Operating system: `Linux a2432sds 5.9.14-arch1-1 #1 SMP PREEMPT Sat, 12 Dec 2020 14:37:12 +0000 x86_64 GNU/Linux`
- Contents of skaffold.yaml:

```yaml
apiVersion: skaffold/v2beta10
kind: Config

build:
  artifacts:
  - context: docker/aria2
    docker:
      buildArgs:
        ARIA2_VERSION: 1.35.0
      noCache: true
    image: equalos/aria2
  tagPolicy:
    envTemplate:
      template: alpine

metadata:
  name: aria2

profiles:
- activation:
  - command: build
    kubeContext: docker-desktop
  build:
    local:
      useBuildkit: true
  name: docker-desktop
```

### Steps to reproduce the behavior

1. Build an image locally using docker-desktop context
2. `skaffold build`

**Comment by othercorey** [comment]
I had to revert to 1.16.0 to get my images to push to remote repository at all.

**Comment by briandealwis** [comment]
Hi @othercorey.  Do you have any settings for `local-cluster` in your `~/.skafold/config` file?  Could you include the output of running with `-v debug`?

There were some changes to this with #5012, but when I run with `docker-desktop`, I see push-images being set to false:
```
...
INFO[0000] applying profile: docker-desktop             
...
INFO[0000] Using kubectl context: docker-desktop        
DEBU[0000] Using builder: local                         
...
DEBU[0000] push value not present, defaulting to false because cluster.PushImages is false 
```

**Comment by braderhart** [comment]
@briandealwis  Check this out:

```
INFO[0000] Activated kube-context "docker-desktop"
DEBU[0000] Defaulting deploy type to kubectl
INFO[0000] Using kubectl context: docker-desktop
DEBU[0000] Using builder: local
DEBU[0000] setting Docker user agent to skaffold-v1.17.2
DEBU[0000] push value not present, defaulting to true because cluster.PushImages is true
```

**Comment by braderhart** [comment]
Because of this I also believe it is ignoring `useBuildkit: true` in my example above. To confirm I would likely need to downgrade to v1.6.0 and see if there is anything in the `-v debug`, but I don't see anything in the current `-v debug` indicating it is being used and the stdout from the build has changed significantly from v1.6.0 that leads me to that conclusion.

**Comment by gsquared94** [comment]
@braderhart  I'm wondering if the profile isn't being applied correctly. Can you try embedding this `build` stanza directly in the main pipeline instead of using a profile and see if you still encounter this issue?
```yaml

apiVersion: skaffold/v2beta10
kind: Config

build:
  local:
    useBuildkit: true
  artifacts:
  - context: docker/aria2
    docker:
      buildArgs:
        ARIA2_VERSION: 1.35.0
      noCache: true
    image: equalos/aria2
  tagPolicy:
    envTemplate:
      template: alpine

metadata:
  name: aria2
  ```

**Comment by briandealwis** [comment]
And can you include your `~/.skaffold/config` too please?

**Comment by briandealwis** [comment]
@braderhart do you have any other contexts defined?

**Comment by braderhart** [comment]
@briandealwis Sorry for the delay. I've been working on some other things so I'm finally circling back around to this. Here is an example of a complete skaffold.yaml:

```
apiVersion: skaffold/v2beta12
kind: Config
metadata:
  name: hugo
build:
  artifacts:
    - image: sample/hugo
      context: docker/hugo
      docker:
        buildArgs:
          CGO: 0
          FLAGS: -ldflags=-s -ldflags=-w
          HUGO_SHA256: f156c31034f013b11ff19aec09ab4d47fd8689befaa6f58222b48ed970722b4b
          HUGO_VERSION: 0.82.0
        noCache: true
deploy:
  kubeContext: docker-desktop
profiles:
  - name: profile-test
    activation:
      - kubeContext: docker-desktop
        command: build
    build:
      local:
        push: false
        useBuildkit: true
      tagPolicy:
        envTemplate:
          template: alpine
```

Additionally, I have `docker-desktop` set as a local-cluster in my ~/.skaffold/config:

```
lobal:
  update-check: false
  survey:
    last-prompted: "2021-02-16T12:46:47-08:00"
  collect-metrics: false
kubeContexts:
- kube-context: docker-desktop
  local-cluster: true
```

You can see here that it gets the correct context but doesn't apply the profile based on it:

skaffold build --verbosity=info
INFO[0000] Skaffold &{Version:v1.20.0 ConfigVersion:skaffold/v2beta12 GitVersion: GitCommit:c48e97690d8daffd68141c2a68fcbe3df6f6936a GitTreeState:clean BuildDate:2021-03-13T14:01:28Z GoVersion:go1.16.2 Compiler:gc Platform:linux/amd64}
INFO[0000] Loaded Skaffold defaults from "/home/_/.skaffold/config"
INFO[0000] Activated kube-context "docker-desktop"
INFO[0000] Using kubectl context: docker-desktop
Generating tags...
 - sample/hugo -> sample/hugo:2c5006f-dirty
INFO[0000] Tags generated in 6.061768ms
Checking cache...
 - sample/hugo: Not found. Building
INFO[0000] Cache check completed in 20.006396ms
Building [sample/hugo]...
Sending build context to Docker daemon  3.072kB

However, if I specify the `kube-context` during build then it applies the profile successfully:

```
skaffold build --verbosity=info --kube-context=docker-desktop
INFO[0000] Skaffold &{Version:v1.20.0 ConfigVersion:skaffold/v2beta12 GitVersion: GitCommit:c48e97690d8daffd68141c2a68fcbe3df6f6936a GitTreeState:clean BuildDate:2021-03-13T14:01:28Z GoVersion:go1.16.2 Compiler:gc Platform:linux/amd64}
INFO[0000] Loaded Skaffold defaults from "/home/_/.skaffold/config"
INFO[0000] applying profile: profile-test
INFO[0000] Activated kube-context "docker-desktop"
INFO[0000] Using kubectl context: docker-desktop
Generating tags...
 - sample/hugo -> sample/hugo:alpine
INFO[0000] Tags generated in 141.087µs
Checking cache...
 - sample/hugo: Not found. Building
INFO[0000] Cache check completed in 16.022578ms
Building [sample/hugo]...
[+] Building 3.4s (6/9)
 => [internal] load build definition from Dockerfile                                               0.1s
 => => transferring dockerfile: 1.16kB                                                             0.0s
 => [internal] load .dockerignore                                                                  0.1s
 => => transferring context: 2B
```

**Comment by tejal29** [comment]
hmm looks like when you specify the `kube-context`, we grab the `local-cluster` bool and detect its a local cluster.
```
kubeContexts:
- kube-context: docker-desktop
  local-cluster: true
```
May be there is a bug on how we apply profiles.

**Comment by gsquared94** [comment]
seems like a dupe of https://github.com/GoogleContainerTools/skaffold/issues/5334

**Comment by nkubala** [comment]
@braderhart can you confirm this is still an issue on the latest version of skaffold (or any version later than v1.27.0)? it seems that this has been fixed in https://github.com/GoogleContainerTools/skaffold/pull/6024 but I want to be sure before closing this issue.

**Comment by braderhart** [comment]
@nkubala It looks like it is working now. Thanks!

**Comment by braderhart** [comment]
Fixed by #6024
